### DISCUSSION TITLE
Unregister nested values with "shouldUnregister: false"

### DISCUSSION THREAD

**Discussion by michaellutsenko** [react-hook-form/react-hook-form#8765] [Q&A] [answered]
I'm using react-hook-form@6.15.1, calling useForm with `shouldUnregister: false` for the purposes of the form I'm working on. Calling unregister on root or nested fields doesn't seem to do anything, the values remain in the form and are included in the submit payload. What would be the correct approach to do so?
Here's a [codesandbox](https://codesandbox.io/s/funny-shadow-193d1p) where I recreated the issue: you'll see than when clicking "submit", both values are included and nested in the "root" object. If you click "unregister" and then "submit" no change is made.
Thanks in advance

**Comment by Moshyfawn** [comment, marked as answer]
From [RHF docs](https://react-hook-form.com/api/useform/unregister/#rules):

> Make sure you unmount that input which has register callback or else the input will get registered again.

**Reply by Moshyfawn** [reply]
`unregister` doesn't remove the form field values, but rather unregisters the field's reference. If you call `unregister` and the form re-renders, the `register` method's gonna be called again, making the field registered again, hence the field's value appears again.

**Reply by michaellutsenko** [reply]
This was so obvious, yet somehow I missed it
Thank you very much, this seems to be the answer

**Reply by Moshyfawn** [reply]
Awesome, no worries, I'm glad it's clear now!
